src/cryptsetup/cryptsetup.c

```diff
@@ -136,10 +144,19 @@ static const char* const passphrase_type_table[_PASSPHRASE_TYPE_MAX] = {
         [PASSPHRASE_BOTH] = "passphrase or recovery key",
 };
 
-const char* passphrase_type_to_string(PassphraseType t);
-PassphraseType passphrase_type_from_string(const char *s);
+static const char* passphrase_type_to_string(PassphraseType t);
+
+DEFINE_PRIVATE_STRING_TABLE_LOOKUP_TO_STRING(passphrase_type, PassphraseType);
+
+static const char* const token_type_table[_TOKEN_TYPE_MAX] = {
+        [TOKEN_TPM2] = "tpm2",
+        [TOKEN_FIDO2] = "fido2",
+        [TOKEN_PKCS11] = "pkcs11",
+};
+
+static const char* token_type_to_string(TokenType t);
```

Comment:
as above, this line is unnecessary, the macro below should take care of everything, no need to define a prototype for this.